sources/engine/Stride.Input/Windows/RawInput/RawInput.cs

```diff
@@ -138,7 +138,7 @@ public static bool RegisterDevice(UsagePage usagePage, UsageId usageId, ModeFlag
             }
             var buffer = stackalloc byte[(int)cbSize];
             var count = Win32.GetRawInputData(lParam, (uint)RawInputDataType.RID_INPUT, (IntPtr)buffer, ref cbSize, (uint)sizeof(RawInputHeader));
-            return (RawInputData*)buffer;
+            return *(RawInputData*)buffer;
```

Comment:
Nice catch @Ethereal77 
You're still returning something pointing to stackalloc'ed data, that data returns to the stack as soon as it exits the scope so if anything re-allocates on the stack it'll manipulate what's in that pointer. Allocate from the caller and pass in the pointer to that function instead to read it from the caller safely

Reply:
No. Because it is a struct it will make a copy(I checked the memory addresses to see if they were different). However, I just realized this will break for the HID inputs because that is a byte pointer in the provided struct. So I will need to change some more stuff.

Reply:
Ah nevermind, somehow swapped diffs around